### PR TITLE
updated stack readme.md file

### LINKED ISSUE #4676: Update readme.md of stack data structure
### What would you like to share?

I want to update the stack readme.md file by introducing declaration of stack and some real life applications of stack data structure.
Also I want to update the defination of Stack.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by AnujRathour**
Fixes #4676

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.

### CODE REVIEW

**Comment by AnujRathour** [comment]
@siriak plese review this pull request and requested to merge under hacktoberfest

File: src/main/java/com/thealgorithms/datastructures/stacks/README.md [hunk 1]
```diff
@@ -1,10 +1,14 @@
 # STACK
 
-Stack is an ADT (abstract data type) that acts like a list of objects but there is a difference.
+- Stack is an ADT (abstract data type) that has a collection of elements where items are added and removed from the end, known as the "top" of the stack.
```
Comment (siriak) [inline src/main/java/com/thealgorithms/datastructures/stacks/README.md:3]:
```suggestion
- Stack is an ADT (abstract data type) that is a collection of elements where items are added and removed from the end, known as the "top" of the stack.
```

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Comment by AnujRathour** [comment]
@siriak thanks for appreciation please add hacktoberfest accepted label.

**Pull request merged**